lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -320,7 +320,13 @@ export class ResponseHandler {
         if (cacheRecord.accessToken) {
             if (cacheRecord.accessToken.tokenType === AuthenticationScheme.POP) {
                 const popTokenGenerator: PopTokenGenerator = new PopTokenGenerator(cryptoObj);
-                accessToken = await popTokenGenerator.signPopToken(cacheRecord.accessToken.secret, request);
+                const { secret, keyId } = cacheRecord.accessToken;
+
+                if (!keyId) {
+                    throw ClientAuthError.createKeyIdMissingError();
```

Comment:
What would be the way an application would handle this? Retry the request?

Reply:
Yes, I don't expect this error to be hit but since keyId is an optional param there has to be a check before passing it into the signPopToken API. 

In order for this error to throw, either the keyId would have to be deleted from the cached access token credential somehow or (in the native client case) the keyId would have to be removed from the NativeRequest.

Any suggestions on how to avoid a check on an optional attribute that isn't optional down the line?

Reply:
No, I think this should be fine.